src/scripts/rx-data-table.src.js

```diff
@@ -434,31 +435,36 @@ app.directive('rxDataTable', function ($http, $timeout, $document, $filter, $par
-                return _.filter(column.icon, function (icon) {
-                    if (_.has(icon, 'fieldValue')) {
-                        if (_.isArray(this.row[icon.field]) && _.contains(this.row[icon.field], icon.fieldValue)) {
+                if( _.has(column, 'icon')) {
+                    return _.filter(column.icon, function (icon) {
+                        if (_.has(icon, 'fieldValue')) {
+                            if (_.isArray(row[icon.field]) && _.includes(row[icon.field], icon.fieldValue)) {
+                                return true;
+                            } else if (icon.fieldValue === row[icon.field]) {
+                                return true;
+                            }
+                        }
+
+                        if (_.has(icon, 'fieldValues')) {
+                            return _.isArray(icon.fieldValues) && _.includes(icon.fieldValues, row[icon.field]);
+                        } else if (row[icon.field] === true) {
                             return true;
-                        } else if (icon.fieldValue === this.row[icon.field]) {
+                        } else if (_.has(icon, 'fieldMinLength') && _.isArray(row[icon.field])) {
+                            return row[icon.field].length >= icon.fieldMinLength;
+                        } else if (icon.persistent === true) {
                             return true;
+                        } else if (_.has(icon, 'editable') && icon.editable === true) {
+                            return !_.isUndefined(column.editable) && column.editable.clause(row);
                         }
-                    }
-
-                    if (_.has(icon, 'fieldValues')) {
-                        return _.isArray(icon.fieldValues) && _.contains(icon.fieldValues, this.row[icon.field]);
-                    } else if (row[icon.field] === true) {
-                        return true;
-                    } else if (_.has(icon, 'fieldMinLength') && _.isArray(row[icon.field])) {
-                        return row[icon.field].length >= icon.fieldMinLength;
-                    } else if (icon.persistent === true) {
-                        return true;
-                    } else if (_.has(icon, 'editable') && icon.editable === true) {
-                        return !_.isUndefined(column.editable) && column.editable.clause(row);
-                    }
-                }, {row: row}).filter(function (icon) {
-                    if ((_.has(icon, 'name') && (this.type === 'i'))||(_.has(icon, 'class') && (this.type === 'div'))) {
-                        return true;
-                    }
-                }, {type: type});
+                    }).filter(function (icon) {
+                        if ((_.has(icon, 'name') && (type === 'i'))||(_.has(icon, 'class') && (type === 'div'))) {
+                            return true;
+                        }
+                    });
+                }
+                else {
```

Comment:
This false is to return that there are no icons available. There should likely be another else  at the end of the above if